pipelines/srp-pipeline.yml

```diff
@@ -0,0 +1,522 @@
+                srp provenance schematic --verbose --path ./metric-store-ci/srp-scan/metric-store/srp_schematic.yaml
+
+                srp provenance compile --saveto ${SRP_WORKING_DIR}/prov3_fragment.json
+
+                echo "Created ${SRP_WORKING_DIR}/prov3_fragment_${BUILD_ID}.json"
+
+                cat ${SRP_WORKING_DIR}/prov3_fragment.json
+                echo
+
+                if [ "${SRP_CLIENT_ID}" != "" ] && [ "${SRP_CLIENT_SECRET}" != "" ]; then
+                  srp provenance submit --verbose --path ${SRP_WORKING_DIR}/prov3_fragment.json
+                else
+                  echo "Skipping SRP submission.  Both SRP_CLIENT_ID and SRP_CLIENT_SECRET must be defined"
+                fi
+
+                srp metadata status \
+                  --srp-endpoint $SRP_URL
+
+  - name: "collect-srp-source-provenance-cf-auth-proxy"
+    serial: true
+    plan:
+      - in_parallel:
+          - get: srp-helper-test
+          - put: build-metadata
+          - get: metric-store-version
+          - get: metric-store-develop
+          - get: metric-store-ci
+      - task: "submit-metadata-for-metric-store-cf-auth-proxy"
+        image: srp-helper-test
+        config:
+          platform: linux
+          inputs:
+            - name: metric-store-ci
+            - name: metric-store-develop
+            - name: build-metadata
+            - name: metric-store-version
+          params:
+            SRP_CLIENT_ID: ((phoenix-osspi.srp-client-id))
+            SRP_CLIENT_SECRET: ((phoenix-osspi.srp-client-secret))
+            SRP_URL: https://srp.vmware.com
+            SRP_KEY: ((phoenix-osspi.svcboteos_github_private_key))
+          run:
+            path: /bin/bash
+            args:
+              - -c
+              - |
+                set -x
+                set -euo pipefail
+                mkdir ./provenance
+
+                VERSION=$(cat metric-store-version/version)
+                BUILD_NUMBER=$(cat build-metadata/build-name)
+                BUILD_PIPELINE_NAME=$(cat build-metadata/build-pipeline-name)
+                BUILD_JOB_NAME=$(cat build-metadata/build-job-name)
+                BUILD_ID=$(cat build-metadata/build-id)
+                OUTPUT_VERSION="1.0.0-${BUILD_ID}"
+                BUILD_PROJECT=metric-store-release
+                BUILD_BRANCH=$(cat metric-store-develop/.git/refs/heads/master)
+                ARTIFACTORY_AUTH_TOKEN=cmVmdGtuOjAxOjE3MjYyMjY1MDI6cWU2U1FlazVjNjVoNUllV2pWWU1BQ0NnaHJV
+
+                export SRP_WORKING_DIR=/srp-work
+                # download Observer agent
+                export PATH=/srp-tools:/srp-tools/observer/bin:$PATH
+                export SRPCLIVERSION="0.15.1-20231012115959-17ea2c6-242"
+                curl -o /tmp/srp-tools-linux.tar.gz  -L https://artifactory.eng.vmware.com/artifactory/srp-tools-generic-local/all/${SRPCLIVERSION}/srp-tools-linux-amd64-${SRPCLIVERSION}.tar.gz &&
+                mkdir -p /srp-tools/observer
+                cd /srp-tools
```

Comment:
instead of  using the "cd" command better to use pushed $dirname popd command

Reply:
Done